src/config/configStore.ts

```diff
@@ -185,8 +261,12 @@ export abstract class BaseConfigStore<T extends BaseConfigStore.Options>
    *
    * @param contents The contents.
    */
-  public setContents(contents?: ConfigContents): void {
-    this.contents = contents || {};
+  public setContents(contents?: P): void {
```

Comment:
I think you can add a default value and it'll inherit the type, e.g.

`setContents(contents: P = {})`